From 82cf89deb997fd2633cbdeaa659a91f51d6e4cae Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Sat, 9 Aug 2014 15:17:44 +0100
Subject: [PATCH] Avoid "pkg-config --atleast-version"

AM_PATH_GTK_3_0([3.4], [], [], [gthread]) results in calling
"pkg-config --atleast-version 3.4 gtk+-3.0 gthread-2.0".

With pkg-config 0.28, that test fails, because the version of
gthread-2.0 is smaller than 3.4. This appears to be a behaviour
change since 0.26, but matches what the documentation said in 0.26.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=641638
Bug-Debian: https://bugs.debian.org/756476
Reviewed-by: Matthias Clasen <mclasen@redhat.com>
Applied-upstream: 3.12.3
---
 m4macros/gtk-3.0.m4 | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/m4macros/gtk-3.0.m4 b/m4macros/gtk-3.0.m4
index 5238b43..0d605fb 100644
--- a/m4macros/gtk-3.0.m4
+++ b/m4macros/gtk-3.0.m4
@@ -11,8 +11,9 @@ dnl Get the cflags and libraries from pkg-config
 dnl
 AC_ARG_ENABLE(gtktest, [  --disable-gtktest       do not try to compile and run a test GTK+ program],
 		    , enable_gtktest=yes)
+  min_gtk_version=ifelse([$1], [], [3.0.0], [$1])
 
-  pkg_config_args=gtk+-3.0
+  pkg_config_args="gtk+-3.0 >= $min_gtk_version"
   for module in . $4
   do
       case "$module" in
@@ -38,7 +39,6 @@ AC_ARG_ENABLE(gtktest, [  --disable-gtktest       do not try to compile and run
     no_gtk=yes
   fi
 
-  min_gtk_version=ifelse([$1], ,3.0.0,$1)
   AC_MSG_CHECKING(for GTK+ - version >= $min_gtk_version)
 
   if test x$PKG_CONFIG != xno ; then
@@ -48,7 +48,7 @@ AC_ARG_ENABLE(gtktest, [  --disable-gtktest       do not try to compile and run
 	  enable_gtktest=no
     fi
 
-    if $PKG_CONFIG --atleast-version $min_gtk_version $pkg_config_args; then
+    if $PKG_CONFIG $pkg_config_args; then
 	  :
     else
 	  no_gtk=yes
@@ -202,10 +202,11 @@ AC_DEFUN([GTK_CHECK_BACKEND],
 [
   pkg_config_args=ifelse([$1],,gtk+-3.0, gtk+-$1-3.0)
   min_gtk_version=ifelse([$2],,3.0.0,$2)
+  pkg_config_args="$pkg_config_args >= $min_gtk_version"
 
   AC_PATH_PROG(PKG_CONFIG, [pkg-config], [AC_MSG_ERROR([No pkg-config found])])
 
-  if $PKG_CONFIG --atleast-version $min_gtk_version $pkg_config_args ; then
+  if $PKG_CONFIG $pkg_config_args ; then
     target_found=yes
   else
     target_found=no
-- 
2.1.0.rc1

